### ISSUE TITLE
Question about the data augmentation

### ISSUE THREAD

**Issue by Tao-Zhou** [zhixuhao/unet#64]
in the dataPrepare.ipynb file, there are the code that
`#you will see 60 transformed images and their masks in data/membrane/train/aug
num_batch = 3
for i,batch in enumerate(myGenerator):
    if(i >= num_batch):
        break`
my question is that how can get 60 transformed images? In  my computer, I get at least 120 images, what's the reason? Could anyone help me?
Thanks

**Comment by ajithvallabai** [comment]
@Tao-Zhou 
i think its from 
`https://github.com/zhixuhao/unet/blob/master/dataPrepare.ipynb `
`
image_arr,mask_arr = geneTrainNpy("data/membrane/train/aug/","data/membrane/train/aug/")
#np.save("data/image_arr.npy",image_arr)
#np.save("data/mask_arr.npy",mask_arr)
`
The above lines first 60 are results of data augumentation done by zhixuhao if you run the code again you will get another 60 images .

**Comment by BarryBA** [comment]
@Tao-Zhou @ajithvallabai @zhixuhao 
I used the code of data augmentation (https://github.com/zhixuhao/unet/blob/master/data.py) to visualize the augmentated results. After definiting the functions, I use the main function to carry out the data augmentation. The code is :
> if __name__ == "__main__":
> # define the data generator
> data_gen_args = dict(rotation_range=0.2, width_shift_range=0.05, height_shift_range=0.05, shear_range=0.05, zoom_range=0.05, horizontal_flip=True, fill_mode='nearest')  
> myGenerator = trainGenerator(batch_size=20, train_path='/home/.../unet/my_unet_keras/',  image_folder='train', mask_folder='label', aug_dict=data_gen_args, 
> save_to_dir='/home/.../unet/my_unet_keras/train_label_aug/')
> # visualize the data augmentation result
> num_batch=3
> for i,batch in enumerate(myGenerator):
> if(i>num_batch):
> break
But the total results is more than 110000 images, and I don't know why the 'mask' images is from 'mask_0_..' to 'mask_30..', which means one more of labels than the number of train images.
Can you tell me why? Is there some thing wrong in the code? Thanks!

**Comment by jizhang02** [comment]
Hello, 
I also have doubts about data augmentation. How can I know the exact number of augmented data or the rules of it?
For example, my train data has 60 images, the batchsize is 2, epochs is 2. But after augmented, I create another directory to see the augmented results. The number of augmented images is 26. And it will change with different epochs, if epoch is 1, the number of augmented images is 24. What the reason for that? 

Could someone explain that?
Thanks a lot!

**Comment by saqibshakeel035** [comment]
> in the dataPrepare.ipynb file, there are the code that
> #you will see 60 transformed images and their masks in data/membrane/train/aug
> ` num_batch = 3 for i,batch in enumerate(myGenerator): if(i >= num_batch): break`
> my question is that how can get 60 transformed images? In my computer, I get at least 120 images, what's the reason? Could anyone help me?
> Thanks

Out of 120 60 are augmented images and 60 are their corresponding masks.
I hope this helps!

**Comment by Drenched-l** [comment]
> 您好，
> 我对数据扩充也有疑问。我怎么知道扩充数据的确切数量或其规则？
> 例如，我的火车数据有60张图像，批处理大小为2，历元为2。但是在扩充后，我创建了另一个目录以查看扩充结果。增强图像的数量为26。并且它将随着不同的时期而变化，如果epoch为1，则增强图像的数量为24。这是什么原因呢？
> 
> 有人可以解释吗？
> 非常感谢！

Hello, what I tested is different from yours. If there is a training set and epoch=2, it should be 60*2=120. After saving, it should be 120*2=240.
